source/contracts/reporting/Auction.sol

```diff
@@ -0,0 +1,289 @@
+            return 0;
+        }
+        return initialAttoRepBalance.sub(_attoRepSold);
+    }
+
+    function getCurrentAttoEthBalance() public returns (uint256) {
+        uint256 _ethSalePriceInAttoRep = getEthSalePriceInAttoRep();
+        uint256 _repSupply = ethAuctionToken.totalSupply();
+        uint256 _attoEthSold = _repSupply.mul(10**18).div(_ethSalePriceInAttoRep);
+        if (_attoEthSold >= initialAttoEthBalance) {
+            return 0;
+        }
+        return initialAttoEthBalance.sub(_attoEthSold);
+    }
+
+    function getDerivedRepPriceInAttoEth() public view returns (uint256) {
+        if (repAuctionToken == IAuctionToken(0) || ethAuctionToken == IAuctionToken(0)) {
+            return repPrice;
+        }
+        uint256 _repAuctionTokenMaxSupply = repAuctionToken.maxSupply();
+        uint256 _ethAuctionTokenMaxSupply = ethAuctionToken.maxSupply();
+        if (_repAuctionTokenMaxSupply == 0 || _ethAuctionTokenMaxSupply == 0) {
+            return repPrice;
+        }
+        uint256 _upperBoundRepPrice = repAuctionToken.maxSupply().mul(10**18).div(initialAttoRepBalance);
+        uint256 _lowerBoundRepPrice = initialAttoEthBalance.mul(10**18).div(_ethAuctionTokenMaxSupply);
+        return _upperBoundRepPrice.add(_lowerBoundRepPrice) / 2;
+    }
+
+    function getRepPriceInAttoEth() public view returns (uint256) {
+        // Use the manually provided REP price if the Auction system is not used for setting fees yet
+        if (!controller.useAuction()) {
+            return manualRepPriceInAttoEth;
+        }
+
+        // If this auction is over and it is a recorded auction use the price it found
+        if (getAuctionIndexForCurrentTime() != currentAuctionIndex && currentRoundType == RoundType.RECORDED) {
+            return getDerivedRepPriceInAttoEth();
+        }
+
+        return repPrice;
+    }
+
+    function getAuctionIndexForCurrentTime() public view returns (uint256) {
+        return controller.getTimestamp().sub(initializationTime) / Reporting.getAuctionDuration();
+    }
+
+    function isActive() public view returns (bool) {
+        RoundType _roundType = getRoundType();
+        return _roundType == RoundType.UNRECORDED || _roundType == RoundType.RECORDED;
+    }
+
+    function getRoundType() public view returns (RoundType) {
+        uint256 _auctionDay = getAuctionIndexForCurrentTime() % 7;
+        return RoundType(_auctionDay);
+    }
+
+    function getAuctionStartTime() public view returns (uint256) {
+        uint256 _auctionIndex = getAuctionIndexForCurrentTime();
+        uint256 _auctionDay = _auctionIndex % 7;
+        uint256 _weekStart = initializationTime.add(_auctionIndex).sub(_auctionDay);
+        uint256 _addedTime = _auctionDay > uint256(RoundType.UNRECORDED) ? uint256(RoundType.RECORDED) : uint256(RoundType.UNRECORDED);
+        return _weekStart.add(_addedTime.mul(Reporting.getAuctionDuration()));
+    }
+
+    function getAuctionEndTime() public view returns (uint256) {
```

Comment:
Might want to consider making the code for this and `getAuctionStartTime` more DRY since they're almost the same. (Then again, it's probably a bit more readable this way.)